### PR TITLE
Add tests to PR #84

### PR DESCRIPTION
**Pull request by astorije**
64 tests before, 112 tests after (**48 new tests**). Most of them are dead simple but that will help detect issues when refactoring small or big portions.

The scary part is the code coverage: it is marked as **98%** at the moment, while it will go down to **62%** when this PR is merged. However, do not get fooled, the overall coverage *does* increases with this PR.
This comes from the fact that this PR asks [Istanbul](https://github.com/gotwarlost/istanbul) to cover all source files where it used to take into account only files that were traversed by the tests. It means that source files that were absolutely not tested... were not counted! Not very accurate.

A question for @tripu: should we add `app.js` to the files ignored by the coverage? On one hand it seems like cheating, on the other hand we will *never* be able to test it using [Mocha](http://mochajs.org/) and our test suite...

### CODE REVIEW

File: .jshintignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ Gruntfile.js
 config.js
 test/views
 test/test.js
```
Comment (astorije) [inline .jshintignore]: This line will be removed when tests are properly split as per #147.

**Comment by astorije** [comment]
@tripu, @darobin, this PR doesn't affect code, just tests, and the ones who failed in the last Travis CI test are untouched... Do you know if something changed on the Specberus side... or maybe the CSS validator side?

**Comment by tripu** [comment]
@astorije, see #159. It's a PR over this other branch of yours (sorry, I just realised it probably should be a normal PR into `master`&hellip;).

If you merge that one, and that fixes the build over here (as I expect), I'll merge this PR in turn :¬)

**Comment by astorije** [comment]
@tripu, I'd say this PR can now be merged. I updated the original description to give you some food for thoughts.

**Comment by astorije** [comment]
@tripu? Pleaaaase :-)
`master` will be broken until merging :-(

**Comment by astorije** [comment]
\o/ Thanks tripu!

**Comment by tripu** [comment]
About code coverage: @astorije, I'd say we include as much as we can&nbsp;&mdash;&nbsp;even code that is impractical or impossible to test. That's the only way for the metric to be really meaningful, isn't it? Better not to fool ourselves with distorted figures. Just an idea :¬)

**Pull request merged**